lib/elixir/lib/process.ex

```diff
@@ -293,8 +293,9 @@ defmodule Process do
   which is looked up at the time of delivery. No error is produced if the name does
   not refer to a process.
 
-  This function returns a timer reference, which can be read with `read_timer/1`
-  or canceled with `cancel_timer/1`.
+  This function returns a timer reference and it's not executed immediately,
```

Comment:
I mean that things can be executed before, I don't know how to phrase it better. The example:

```
iex(1)> pid = self(); Process.send_after(pid, :foo, 0); send(pid, :bar); receive do; any -> IO.inspect(any); end
:bar
:bar
```

Reply:
That's all the standard beam messaging system, it will get sent when the send_after process wakes up to send the message.  All messages have an inherent delay (as in the real world, of which it was modeled after for it's original usecase), but the send_after process itself has no wait on the 0 whatsoever and this phrasing seems to imply it does?

Reply:
This behaviour is different from using the `send`

```
iex(1)> pid = self(); send(pid, :foo); send(pid, :bar); receive do; any -> any; end
:foo
```

I believe that some people would think that `Process.send_after(pid, :message, 0)` is equivalent to `send(pid, :message)` but it's not, I wanted to clarify that.

Reply:
> This behaviour is different from using the `send`

That's because your code is not equivalent functionality, this would more closely be:
```elixir
iex(1)> pid = self(); spawn(fn -> send(pid, :foo) end); send(pid, :bar); receive do any -> any after 1 -> :failed end
:bar
```

With `Process.send_after/4` you are asking 'another' process to send you back a message, which will always be slower than just sending yourself a message.

Reply:
I know, but the documentation doesn't say anything about it. I found one of this causing problems in a project that I work, I believe that the behaviour is not clear to everyone

Reply:
Should document that it's sent through another process then.  It is *entirely* possible for your process to get pre-empted and the other to run before you run again, thus changing the order of the above examples.  You cannot rely on messaging ordering across different process messaging, and thus stating that you can run other code before you receive the message is not always true.